app/views/govuk_publishing_components/components/_input.html.erb

```diff
@@ -84,17 +90,23 @@
   <% end %>
 
   <% if hint %>
+    <% hint_rtl = label_rtl.nil? ? right_to_left : label_rtl %>
+
     <%= render "govuk_publishing_components/components/hint", {
       id: hint_id,
-      text: hint
+      text: hint,
+      right_to_left: hint_rtl
     } %>
   <% end %>
 
   <% if has_error %>
+    <% error_rtl = label_rtl.nil? ? right_to_left : label_rtl %>
```

Comment:
Asking mainly for my own understanding (it's early, my brain's still waking up) - the logic for hint and error differs from the logic from label? What's the difference there?

Reply:
This is now outdated and improved by the later changes along the lines suggested by Kevin.